crates/bevy_color/src/color.rs

```diff
@@ -0,0 +1,128 @@
+use crate::{Hsla, Lcha, LinearRgba, Oklaba, Srgba};
+
+/// An enumerated type that can represent any of the color types in this crate.
+///
+/// This is useful when you need to store a color in a data structure that can't be generic over
+/// the color type.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Color {
+    /// A color in the sRGB color space with alpha.
+    Srgba(Srgba),
+    /// A color in the linear sRGB color space with alpha.
+    LinearRgba(LinearRgba),
+    /// A color in the HSL color space with alpha.
+    Hsla(Hsla),
+    /// A color in the LCH color space with alpha.
+    Lcha(Lcha),
+    /// A color in the Oklaba color space with alpha.
+    Oklaba(Oklaba),
+}
+
+impl Color {
```

Comment:
I suspect if we swap our current Color type for this, that we will want to add a bunch of convenience constructors here. But that can wait!